xonsh/history/sqlite.py

```diff
@@ -220,6 +221,17 @@ def xh_sqlite_wipe_session(sessionid=None, filename=None):
         c.execute(sql, (str(sessionid),))
 
 
+def xh_sqlite_delete_input_matching(pattern, filename=None):
+    """Deletes entries from the database where the input matches a pattern."""
+    with _xh_sqlite_get_conn(filename=filename) as conn:
+        c = conn.cursor()
+        _xh_sqlite_create_history_table(c)
+        for inp, *_ in _xh_sqlite_get_records(c):
+            if pattern.match(inp):
+                sql = f"DELETE FROM xonsh_history WHERE inp LIKE '{inp}'"
+                c.execute(sql)
```

Comment:
Instead of using like you can use = sign. If the input itself had '%' in it , it might not properly escape it.

Reply:
Great! Thanks. I've changed the code to use the '='-operator instead. :+1: